lib/pure/asyncdispatch.nim

```diff
@@ -542,6 +542,94 @@ when defined(windows) or defined(nimdoc):
       retFuture.fail(newException(OSError, osErrorMsg(lastError)))
     return retFuture
 
+  proc recvInto*(socket: TAsyncFD, buf: cstring, size: int,
+                 flags = {SocketFlag.SafeDisconn}): Future[int] =
+    ## Reads **up to** ``size`` bytes from ``socket`` into ``buf``, which must
+    ## at least be of that size. Returned future will complete once all the
+    ## data requested is read, a part of the data has been read, or the socket
+    ## has disconnected in which case the future will complete with a value of
+    ## ``0``.
+    ##
+    ## **Warning**: The ``Peek`` socket flag is not supported on Windows.
```

Comment:
This function is, in my opinion, extremely dangerous because the ``buf`` param needs to point to data which is allocated until the returned ``Future`` completes. Is there any way we could make this safer?

At the very least I think this deserves a warning in the documentation.